harness/determined/pytorch/_pytorch_context.py

```diff
@@ -338,7 +341,8 @@ def _init_device(self) -> torch.device:
             device = torch.device("cuda", 0)
         else:
             device = torch.device("cpu")
-        check.is_not_none(device)
+        if device is None:
+            raise Exception("Error initializing torch.device")
```

Comment:
This check seems unnecessary to me.  Afaict, there's no branch where it might be None.

Reply:
Removed.